### ISSUE TITLE
Ambiguous bases

### ISSUE THREAD

**Issue by HadrienG** [HadrienG/InSilicoSeq#20]
When genomes are downloaded from the ncbi they sometimes contain ambiguous bases (e.g. 'S' which stands for 'G' or 'C') Those cause a `KeyError` in the dispatch dict for indels

Traceback:

```python
File "/usr/local/Cellar/python3/3.6.2/Frameworks/Python.framework/Versions/3.6/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/local/Cellar/python3/3.6.2/Frameworks/Python.framework/Versions/3.6/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/Users/hadrien/Documents/github/InSilicoSeq/iss/__main__.py", line 5, in <module>
    main()
  File "/Users/hadrien/Documents/github/InSilicoSeq/iss/app.py", line 318, in main
    args.func(args)
  File "/Users/hadrien/Documents/github/InSilicoSeq/iss/app.py", line 123, in generate_reads
    generator.to_fastq(read_gen, args.output)
  File "/Users/hadrien/Documents/github/InSilicoSeq/iss/generator.py", line 124, in to_fastq
    for read_tuple in generator:
  File "/Users/hadrien/Documents/github/InSilicoSeq/iss/generator.py", line 86, in reads
    reverse, 'reverse', sequence, bounds)
  File "/Users/hadrien/Documents/github/InSilicoSeq/iss/error_models/__init__.py", line 179, in introduce_indels
    if random.random() < deletions[position][mutable_seq[nucl]]:  # del
KeyError: 'S'
```

IUPAC codes: http://www.bioinformatics.org/sms2/iupac.html

**Comment by HadrienG** [comment]
fixed in a6cf340

**Comment by HadrienG** [comment]
caught another similar bug in the `mut_sequence` method. fixed in 8ecb2ab

**Comment by simroux** [comment]
Hi, 
Thanks for the great tool, really useful ! It looks like I stumbled across a variant of this bug, although the KeyError doesn't really make sense to me: 
`Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/joblib/externals/loky/process_executor.py", line 418, in _process_worker
    r = call_item()
  File "/usr/local/lib/python3.6/site-packages/joblib/externals/loky/process_executor.py", line 272, in __call__
    return self.fn(*self.args, **self.kwargs)
  File "/usr/local/lib/python3.6/site-packages/joblib/_parallel_backends.py", line 567, in __call__
    return self.func(*args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/joblib/parallel.py", line 225, in __call__
    for func, args, kwargs in self.items]
  File "/usr/local/lib/python3.6/site-packages/joblib/parallel.py", line 225, in <listcomp>
    for func, args, kwargs in self.items]
  File "/usr/local/lib/python3.6/site-packages/iss/generator.py", line 59, in reads
    forward, reverse = simulate_read(record, ErrorModel, i)
  File "/usr/local/lib/python3.6/site-packages/iss/generator.py", line 134, in simulate_read
    forward.seq = ErrorModel.mut_sequence(forward, 'forward')
  File "/usr/local/lib/python3.6/site-packages/iss/error_models/__init__.py", line 100, in mut_sequence
    nucl_choices[position][nucl][0],
KeyError: 'a'
`

I'm running 1.3.1 through Docker (docker:hadrieng/insilicoseq:1.3.1). This error disappear when I remove N's from my input genome, so I suspect it's something to do with how ambiguous base are handled ? (although I'm not sure what the error comes up as lower case a). Note that I got similar KeyError with 't", 'g', and 'c' (all lower case).

Thanks !

**Comment by HadrienG** [comment]
Hi @simroux !

I think your error is more closely related to #73 , which I thought I fixed. I'll reopen the issue and let you know

**Comment by HadrienG** [comment]
Should be fixed in `1.3.3`
